src/main/java/spoon/support/compiler/SpoonPom.java

```diff
@@ -541,15 +535,30 @@ public static String guessMavenHome() {
 					}
 				}
 			}
-
 			p.waitFor();
 		} catch (IOException e) {
 			throw new SpoonException("Maven home detection has failed.");
 		} catch (InterruptedException e) {
 			Thread.currentThread().interrupt();
 			throw new SpoonException("Maven home detection was interrupted.");
 		}
-		return mvnHome;
+		throw new SpoonException("Couldn't find path to maven home.");
+	}
+
+	private static String getPathToMavenExecutable() {
+		String executableName;
+		if (System.getProperty("os.name").contains("Windows")) {
+			executableName = "mvn.cmd";
+		} else {
```

Comment:
Does this work on Mac? You removed the Mac case?

Reply:
Yes. I tested it [here](https://github.com/SpoonLabs/sorald/runs/4309903908?check_suite_focus=true#step:5:8).

1. I get the absolute path to the maven executable as I have done in this PR.
2. Execute `</absolute/path/to/maven> -version`.